consensus/hotstuff/integration/instance_test.go

```diff
@@ -154,23 +159,28 @@ func NewInstance(t require.TestingT, options ...Option) *Instance {
 	// program the builder module behaviour
 	in.builder.On("BuildOn", mock.Anything, mock.Anything).Return(
 		func(parentID flow.Identifier, setter func(*flow.Header) error) *flow.Header {
-			parent, ok := in.headers.Load(parentID)
+			in.updatingBlocks.RLock()
+			defer in.updatingBlocks.RUnlock()
+
+			parent, ok := in.headers[parentID]
 			if !ok {
 				return nil
 			}
 			header := &flow.Header{
 				ChainID:     "chain",
 				ParentID:    parentID,
-				Height:      parent.(*flow.Header).Height + 1,
+				Height:      parent.Height + 1,
 				PayloadHash: unittest.IdentifierFixture(),
 				Timestamp:   time.Now().UTC(),
 			}
 			require.NoError(t, setter(header))
-			in.headers.Store(header.ID(), header)
+			// in.headers[header.ID()] = header
```

Comment:
can we remove this? I think the header is processed locally and stored when we broadcast the proposal